compiler/resolution/resolveFunction.cpp

```diff
@@ -1976,7 +1976,12 @@ void resolveReturnTypeAndYieldedType(FnSymbol* fn, Type** yieldedType) {
     ret->type = retType;
 
     if (retType == dtUnknown) {
-      USR_FATAL(fn, "unable to resolve return type");
+      if (fn->hasFlag(FLAG_COMPUTE_UNIFIED_TYPE_HELP)) {
+        USR_FATAL(callStack.v[callStack.n-2],
+                  "can't compute a unified element type for this array");
+      } else {
+        USR_FATAL(fn, "unable to resolve return type");
+      }
```

Comment:
I started out by trying to incorporate "try in reolve" logic, but since `inTryResolve` and the accompanying state variables are all in functionResolution.cpp rather than resolveFunction.cpp, it wasn't easy.

That said, I'm also not seeing why that check would make the error more principled (or at least, any more principled than what is on `main` today)?